#!/usr/bin/env python3

PACKAGE = "rqt_param_demo"


from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add(
    "linear_speed", double_t, 0,
    "Linear speed of robot",
    0.5,   # default
    0.0,   # min
    2.0    # max
)

gen.add(
    "angular_speed", double_t, 0,
    "Angular speed of robot",
    1.0,
    0.0,
    3.14
)

gen.add(
    "acceleration_limit", double_t, 0,
    "Max acceleration",
    1.0,
    0.0,
    5.0
)

gen.add(
    "enable_motion", bool_t, 0,
    "Enable or disable motion",
    True
)

exit(gen.generate(PACKAGE, "motion_controller", "MotionConfig"))

